Source/GSGormLoading.m

```diff
@@ -223,46 +223,55 @@ - (void) awakeWithContext: (NSDictionary *)context
 	}
       else
 	{
-	  topObjects = nil; 
+	  topObjects = nil;
 	}
 
 
       /*
        * Now tell all the objects that they have been loaded from
-       * a nib.
+       * a gorm model.
        */
       enumerator = [nameTable keyEnumerator];
       while ((key = [enumerator nextObject]) != nil)
 	{
-	  if ([context objectForKey: key] == nil || 
+	  if ([context objectForKey: key] == nil ||
 	      [key isEqualToString: NSNibOwner]) // we want to send the message to the owner
 	    {
-	      // we don't want to send a message to these menus twice, if they're custom classes. 
-	      if ([key isEqualToString: @"NSWindowsMenu"] == NO && 
-		  [key isEqualToString: @"NSServicesMenu"] == NO && 
+	      // we don't want to send a message to these menus twice, if they're custom classes.
+	      if ([key isEqualToString: @"NSWindowsMenu"] == NO &&
+		  [key isEqualToString: @"NSServicesMenu"] == NO &&
 		  [key isEqualToString: NSNibTopLevelObjects] == NO)
 		{
 		  id o = [nameTable objectForKey: key];
 
-		  // send the awake message, if it responds...
-		  if ([o respondsToSelector: @selector(awakeFromNib)])
+		  // send the awake message, every object should respond to this message since
+		  // it is defined on NSObject(NSNibAwaking)
+		  [o awakeFromNib];
+
+		  // Call prepareForInteraceBuilder if we are in IB/Gorm.
+		  if ([self respondsToSelector: @selector(isInInterfaceBuilder)])
 		    {
-		      [o awakeFromNib];
+		      if ([self isInInterfaceBuilder] == YES)
```

Comment:
In the NIB and XIB loading code we make this call on NSClassSwapper, why is it on self here?

Reply:
This is because we do things slightly differently with .gorm files since they were created first.  I learned some lessons and did it a bit more correct in Nib (and that was inherited by Xib since they are related).

Reply:
Don't you think it would be about time to change this? But maybe not in this MR.